Neos.Neos/Classes/FrontendRouting/NodeUri/Options.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Neos\Neos\FrontendRouting\NodeUri;
+
+/*
+ * Immutable filter DTO for {@see NodeUriBuilder::uriFor()}
+ *
+ * Example:
+ *
+ * Options::create(forceAbsolute: true);
+ *
+ * @api for the factory methods; NOT for the inner state.
+ */
+final readonly class Options
```

Comment:
naming this options will make it required to be aliased when we introduce also `Options` for flows uribuilder... but thats why its in a separate namespace `NodeUri` so one can use 

```php
use Neos\Neos\FrontendRouting\NodeUri;

    /**
      * @Flow\Inject
      * @var NodeUri\NodeUriBuilderFactory
      */
     protected $nodeUriBuilderFactory;

NodeUri \Options::create(forceAbsolute: true);
```

should be fine?

Reply:
Aaaand `routingArguments` should probably be deprecated and get its own with'er?

Reply:
on the other hand

```php
use Neos\Neos\FrontendRouting;

    /**
      * @Flow\Inject
      * @var FrontendRouting\NodeUriBuilderFactory
      */
     protected $nodeUriBuilderFactory;

FrontendRouting\Options::create(forceAbsolute: true);
```

might be even simpler

Reply:
@kitsunet @bwaidelich i dont know about our name space philosophy but here the lattter / inlining it to `FrontendRouting` might make the most sense?